rustls/src/conn.rs

```diff
@@ -1356,7 +1331,22 @@ impl Quic {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) enum Side {
+    Client,
+    Server,
+}
+
+pub(crate) use sealed::SealedSideData;
+
 /// Data specific to the peer's side (client or server).
-pub trait SideData {}
+pub trait SideData: SealedSideData {}
+
+mod sealed {
+    use crate::msgs::enums::HandshakeType;
+    pub trait SealedSideData {
+        const REJECT_RENEGOTIATION_TYPE: HandshakeType;
```

Comment:
Since `CommonState::process_main_protocol()`, which uses this value, also needs to instantiate a `Context` with a concrete `Data` type, I think it doesn't really make sense to use a `&dyn` for this.